src/pyobo/constants.py

```diff
@@ -99,3 +102,25 @@
     "isbn",
     "issn",
 }
+
+# Load version pin dictionary from the environmental variable VERSION_PINS
+try:
+    VERSION_PINS_STR = os.getenv("VERSION_PINS")
+    if not VERSION_PINS_STR:
+        VERSION_PINS = {}
+    else:
+        VERSION_PINS = json.loads(VERSION_PINS_STR)
+        for k, v in VERSION_PINS.items():
+            if not isinstance(k, str) or not isinstance(v, str):
+                raise ValueError("The prefix and version name must both be "
+                                 "strings")
+except Exception as e:
+    raise ValueError("The value for the environment variable VERSION_PINS"
```

Comment:
Not sure it's worth raising an error here. Perhaps logging an error and resetting VERSION_PINS = {} would be a better solution here.